static/js/util/profile_edit.js

```diff
@@ -12,72 +16,141 @@ import moment from 'moment';
 
 // bind to this.boundTextField in the constructor of a form component
 // to update text fields when editing the profile
-export function boundTextField(key, label, error) {
-  const { updateProfile, profile } = this.props;
+// we pass in a keyset looking like this:
+//
+// ["top-level-key", index, "nested_object_key"] or just ["top_level_key"]
+export function boundTextField(keySet, label) {
+  const {
+    profile,
+    errors,
+    updateProfile
+  } = this.props;
   let onChange = e => {
     let clone = Object.assign({}, profile);
```

Comment:
This will need to be a deep clone since `_.set` can alter data at any level within the object